src/main/java/Duke.java

```diff
@@ -1,10 +1,241 @@
+     * @return task number.
+     */
+    public static int getTaskNum(String arg) {
+        String[] words = arg.trim().split("[\\s]+");
+        return Integer.parseInt(words[1]);
+    }
+
+    /**
+     * Returns the description of the task only, without the date or the keyword.
+     *
+     * @param query user raw data input.
+     * @return description of task.
+     */
+    public static String getQueryDescription(String query) {
+        String[] words = query.trim().split("[\\s]+");
+        String[] allButFirstWord = Arrays.copyOfRange(words, 1, words.length);
+        StringBuilder sentenceAfterDeletion = new StringBuilder();
+        for (String word : allButFirstWord) {
+            if (word.contains("/")) {
+                break;
+            } else {
+                sentenceAfterDeletion.append(word).append(" ");
+            }
+        }
+        return sentenceAfterDeletion.toString();
+    }
+
+    /**
+     * Returns date value for tasks which need a date input field.
+     * If user does not have proper date formatting, (i.e. '/by' or '/at') this function returns an empty string.
+     *
+     * @param query user raw data input.
+     * @return date value
+     */
+    public static String getDate(String query) {
+        int slashPosition = query.indexOf("/");
+        if (slashPosition == -1) {
+            return "";
+        } else {
+            int datePosition = slashPosition + 3;
+            return query.substring(datePosition).trim();
+        }
+    }
+
+    /**
+     * Return void.
+     * Function is responsible for printing out the whole task list of the user.
+     *
+     * @param tasks list of tasks input by user
+     */
+    public static void printList(Task[] tasks) {
+        int count = 0;
+        System.out.println("Here is your list:");
+        for (Task item : tasks) {
+            if (tasks[count] != null) {
+                count++;
+                System.out.println(count + ". " + item);
+            }
+        }
+    }
+
+    public static boolean hasListKeyword(String arg) {
+        return arg.trim().matches("^list$");
+    }
+
+    public static boolean hasDoneKeyword(String arg) {
+        return arg.trim().matches("^[done]+\\s+[0-9]+$");
+    }
+
+    public static boolean hasTodoKeyword(String arg) {
+        return arg.trim().toLowerCase().contains("todo");
+    }
+
+    public static boolean hasDeadlineKeyword(String arg) {
+        return arg.trim().toLowerCase().contains("deadline");
+    }
+
+    public static boolean hasEventKeyword(String arg) {
+        return arg.trim().toLowerCase().contains("event");
+    }
+
+    /**
+     * Returns the required value for keyword which is the first word keyed in by user.
+     *
+     * @param query user raw data input.
+     * @return keyword value from Keyword enum class.
+     */
+    public static Keyword getKeywordStatus(String query) {
+        Keyword keyword;
+        if (hasDoneKeyword(query)) {
```

Comment:
I like how neat the code is here, with the use of constants to make the code readable.